awx/main/access.py

```diff
@@ -1994,40 +1947,34 @@ class UnifiedJobTemplateAccess(BaseAccess):
     '''
 
     model = UnifiedJobTemplate
+    select_related = (
+        'created_by',
+        'modified_by',
+        'next_schedule',
+    )
+    # prefetch last/current jobs so we get the real instance
+    prefetch_related = (
+        'last_job',
+        'current_job',
+        Prefetch('labels', queryset=Label.objects.all().order_by('name')),
+    )
 
-    def get_queryset(self):
-        if self.user.is_superuser or self.user.is_system_auditor:
-            qs = self.model.objects.all()
-        else:
-            qs = self.model.objects.filter(
-                Q(pk__in=self.model.accessible_pk_qs(self.user, 'read_role')) |
-                Q(inventorysource__inventory__id__in=Inventory._accessible_pk_qs(
-                    Inventory, self.user, 'read_role')))
-        qs = qs.exclude(inventorysource__source="")
-
-        qs = qs.select_related(
-            'created_by',
-            'modified_by',
-            'next_schedule',
-        )
-        # prefetch last/current jobs so we get the real instance
-        qs = qs.prefetch_related(
-            'last_job',
-            'current_job',
-            Prefetch('labels', queryset=Label.objects.all().order_by('name'))
-        )
-
-        # WISH - sure would be nice if the following worked, but it does not.
-        # In the future, as django and polymorphic libs are upgraded, try again.
+    # WISH - sure would be nice if the following worked, but it does not.
+    # In the future, as django and polymorphic libs are upgraded, try again.
 
-        #qs = qs.prefetch_related(
-        #    'project',
-        #    'inventory',
-        #    'credential',
-        #    'credential__credential_type',
-        #)
+    #qs = qs.prefetch_related(
+    #    'project',
+    #    'inventory',
+    #    'credential',
+    #    'credential__credential_type',
+    #)
 
-        return qs.all()
+    def filtered_queryset(self):
+        qs = self.model.objects.filter(
+            Q(pk__in=self.model.accessible_pk_qs(self.user, 'read_role')) |
+            Q(inventorysource__inventory__id__in=Inventory._accessible_pk_qs(
+                Inventory, self.user, 'read_role')))
+        return qs.exclude(inventorysource__source="")
```

Comment:
That was a logic bug here. This source should be excluded for superusers as well.